pydicom/filereader.py

```diff
@@ -929,6 +929,103 @@ def data_element_offset_to_value(is_implicit_VR, VR):
     return offset
 
 
+def read_frame(fp, is_little_endian, data_elem_value_offset, frame_number,
+               frame_length=None):
+    """Return all fragments of an individual frame of a multi-frame
+    Pixel Data element.
+
```

Comment:
I might be confused, but don't we already have functions for this in `encaps.py`?

```python
def generate_pixel_data_frame(bytestream):
    """Yield an encapsulated pixel data frame as bytes.
```
At the very least it seems like a fair bit of duplication here

Reply:
> Did you mean to include the pydicom/pixel_data_handlers/__pycache__/*.pyc files?

No! I removed them again. Thanks for picking this. I'm used to automatically excluding them via a `.gitignore` file.

Reply:
> I might be confused, but don't we already have functions for this in encaps.py?

You are right, there are functions for this in `encaps.py`. In particular, `get_frame_offsets()` and `read_item()` could be reused. However, I had problems getting this implemented (see [above comment](https://github.com/pydicom/pydicom/pull/534#discussion_r165067722)). I will look into over the next days again, but more extensive refactoring may be required to prevent circular imports.